Docs: Combines thresholds docs into a single topic (#47985)
grafana/grafana: docs/sources/panels/configure-thresholds/_index.md

weight: 300
aliases:
  [
    docs/grafana/latest/panels/thresholds/,
    docs/grafana/latest/panels/,
    docs/grafana/latest/panels/specify-thresholds/about-thresholds/,
    docs/grafana/latest/panels/specify-thresholds/add-a-threshold/,
    docs/grafana/latest/panels/specify-thresholds/add-threshold-to-graph/,
    docs/grafana/latest/panels/specify-thresholds/delete-a-threshold/,
  ]
---

# Configure visualization thresholds

This section includes information about using thresholds in your visualizations. You'll learn about thresholds and their defaults, how to add or delete a threshold, and adding a threshold to a legacy panel.

## About thresholds

A threshold is a value that you specify for a metric that is visually reflected in a dashboard when the threshold value is met or exceeded.

Thresholds provide one method for you to conditionally style and color your visualizations based on query results. You can apply thresholds to most, but not all, visualizations. For more information about visualizations, refer to [Visualization panels]({{< relref "../../visualizations" >}}).

You can use thresholds to:

> - Color lines or areas in the [Graph panel]({{< relref "../../visualizations/graph-panel" >}})

Reviewer: This should also link to new time series panel, the old Graph panel is deprecated. We do not need to link to it. 

```suggestion
- Color grid lines or grid ares areas in the [Time series visualization]({{< relref "../../visualizations/time-series" >}})
```

- Color lines in the [Time-series panel]({{< relref "../../visualizations/time-series/graph-color-scheme/#from-thresholds" >}})
- Color the background or value text in the [Stat panel]({{< relref "../../visualizations/stat-panel" >}})
- Color the gauge and threshold markers in the [Gauge panel]({{< relref "../../visualizations/gauge-panel" >}})
- Color markers in the [Geomap panel]({{< relref "../../visualizations/geomap" >}})
- Color cell text or background in the [Table panel]({{< relref "../../visualizations/table" >}})
- Define regions and region colors in the [State timeline panel]({{< relref "../../visualizations/state-timeline" >}})

> There are two types of thresholds:

Reviewer: The visualizations are in the visualization docs category (and not the Panels category), and we refer to them as visualizations in the UI, yet here we refer to visualizations with a "panel" suffix.

Author: corrected this, thanks

- **Absolute** thresholds are defined by a number. For example, 80 on a scale of 1 to 150.
- **Percentage** thresholds are defined relative to minimum or maximum. For example, 80 percent.

### Default thresholds

On visualizations that support it, Grafana sets default threshold values of:

- 80 = red
- Base = green
- Mode = Absolute

The **Base** value represents minus infinity. It is generally the “good” color.

## Add or delete a threshold

You can add as many thresholds to a panel as you want. Grafana automatically sorts thresholds values from highest to lowest.

Delete a threshold when it is no longer relevant to your business operations. When you delete a threshold, the system removes the threshold from all visualizations that include the threshold.

**Before you begin:**

> - [Add a panel to a dashboard]({{< relref "../working-with-panels/add-panel.md" >}})

Reviewer: this feels unnecessary, should we also prompt user to "Login", open a dashboard? 

They don't need to add a panel to a dashboard, they likely already have one. 

The sentence before reads  `You can add as many thresholds to a panel as you want`  so it's clear this is in the context of a panel.

Author: The effect we're trying to achieve here is to help users who come through a Google search and land blindly on the page understand that having a panel to work with is a prerequisite to performing this task and to provide a link.

The decision to include a Before you begin section is nuanced and very much driven by our audience. If it is super obvious to our audience (even those who blindly land on the page) that they are looking at these docs and already have a panel, then yes, we should remove this. Your feedback is appreciated and gets me thinking about how we can use the Before you begin section more strategically. I'm all for removing non-value-added content. But in some cases, a Before you begin section can be very useful.

Author: Made this change.

**To add or delete a threshold**:

1. To add a threshold:

>    a. Edit the panel to which you want to add a threshold.

Reviewer: Here it also says you need a panel, (see comment above). This step is good, but makes the comment section of adding a panel unnecessary.

Author: corrected

>    b. On the panel display options, locate the **Thresholds** section and click **+ Add threshold**.

Reviewer: ```suggestion
   b. In the options side pane, locate the **Thresholds** section and click **+ Add threshold**.
```

>    b. On the panel display options, locate the **Thresholds** section and click **+ Add threshold**.

Reviewer: ```suggestion
   b. In the options side pane, locate the **Thresholds** section and click **+ Add threshold**.
```

>    c. Selct a threshold color, number, and mode.

Reviewer: ```suggestion
   c. Select a threshold color, number, and mode.
```

+++
title = "Configure thresholds in a Grafana dasboard"
aliases = ["/docs/grafana/latest/panels/thresholds/", "/docs/grafana/latest/panels/", "/docs/grafana/latest/panels/specify-thresholds/about-thresholds/", "/docs/grafana/latest/panels/specify-thresholds/add-a-threshold/", "/docs/grafana/latest/panels/specify-thresholds/add-threshold-to-graph/", "/docs/grafana/latest/panels/specify-thresholds/delete-a-threshold/"]
weight = 300
+++

# Configure thresholds in a dashboard

This section includes information about using thresholds in your visualizations.

> 

Reviewer 2: @jdbaldry - example of where we'd like to insert a TOC for the page. Not a blocker for this PR, just alerting you to the example.

Reviewer: @GrafanaWriter Every page get's an automatic TOC in the right side pane:
![Screenshot from 2022-04-22 13-47-47](https://user-images.githubusercontent.com/10999/164708711-0863c95f-6896-48af-a2a4-aed81222acce.png)

## About thresholds

> A threshold is a value that you specify for a metric that is visually reflected in a dashboard when the threshold value is met or exceeded.

Author: added these points, thanks.

There are two types of thresholds:

- **Absolute** thresholds are defined based on a number. For example, 80 on a scale of 1 to 150.
- **Percentage** thresholds are defined relative to minimum or maximum. For example, 80 percent.

You can apply thresholds to most, but not all, visualizations. For more information about visualizations, refer to [Visualization panels]({{< relref "../../visualizations" >}}).

### Default thresholds

On visualizations that support it, Grafana sets default threshold values of:

- 80 = red
- Base = green
- Mode = Absolute

The **Base** value represents minus infinity. It is generally the “good” color.

## Add a threshold

You can add as many thresholds to a panel as you want. Grafana automatically sorts thresholds from highest value to lowest.

### Before you begin

> - [Add a panel to a dashboard]({{< relref "../working-with-panels/add-panel.md" >}}).

Author: corrected.

> **To add a threshold**:

Author: removed

1. Edit the panel to which you want to add a threshold.
1. On the panel display options, locate the **Thresholds** section.
1. Click **+ Add threshold**.

   Grafana adds a threshold value and color.

1. Accept the recommendations or edit the threshold.
   - **Edit color:** To select a color, click the color dot.
   - **Edit number:** To change the threshold value, click in the field and enter a number.
1. Select a **Threshold mode**.
   Threshold mode applies to all thresholds on this panel.

> 1. In the **Show thresholds** drop-down, select a threshold display option.

Reviewer: Show thresholds option is only available when you have the time series visualization.  So would move this to a paragraph "Show thresholds in graphs" is something like that

Author: qualified this to point out this is only relevant for time-series viz.

> 1. Click **Save**.

Reviewer: I would remove the save prompt, they are probably doing more things. This feels unrelated to there current workflow

## Delete a threshold

Delete a threshold when it is no longer relevant to your business operations. When you delete a threshold, the system removes the threshold from all visualizations that include the threshold.

### Before you begin

> - [Add a threshold]({{< relref "#add-a-threshold.md" >}}).

Reviewer: This feels unnecessary. A user does not want to delete a threshold if they don't already have one

> **To delete a threshold**:

Reviewer: repeats section title action

Author: ah, we can talk about this. When there are entries in the before you begin section, it's visually jarring to go from a bulleted list to a numbered list. The stem sentence provides a bit of separation between the two. We could have a style that removes the stem sentence if there are no before you begin items.

Author: removed this

1. Navigate to the panel to which you want to add a threshold.

> 1. Click the **Field** tab. (Or **Panel** tab for a graph panel.)

Reviewer: There is no Field tab anymore, was removed in v8

Author: Done.

1. Click the trash can icon next to the threshold you want to remove.

> 1. Click **Save** to save the changes in the dashboard.

Reviewer: Do we need to prompt them to save?

Author: It appears that the user doesn't actually need to save their changes, which I didn't actually test. I can specify a threshold and back out of the Edit view without saving, and the threshold I added remains. i'll remove the Save step.
